server/services/memory/migration_task.py

```
@@ -0,0 +1,157 @@
+                               ON CONFLICT (org_id, title) DO NOTHING
+                               RETURNING id""",
+                            (org_id, user_id, content),
+                        )
+                        row = cursor.fetchone()
+                        if row:
+                            create_version(cursor, str(row[0]), org_id, user_id, content, source="migration")
+                            stats["context"] += 1
+                    except Exception as e:
+                        logger.warning(f"[Migration] Error migrating KB memory for org {org_id}: {e}")
+                        stats["errors"] += 1
+                        conn.rollback()
```

Ruff (0.15.18)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)